test/fixtures/poro.rb

```diff
@@ -222,6 +222,18 @@ def maker
   end
 end
 
+TypedRoleSerializer = Class.new(ActiveModel::Serializer) do
+  type 'my-roles'
+  cache only: [:name], skip_digest: true
+  attributes :id, :name, :description, :slug
+
+  def slug
+    "#{object.name}-#{object.id}"
+  end
```

Comment:
noting that in `rc4` slug won't be call here. you'd have to do `attribute :slug { "#{object.name}-#{object.id}" }`.  can be removed from the poro here, though, since it's not being used